js/test/daterange/testdatefmtrange_fmttemplate.js

```diff
@@ -28,36 +28,38 @@ module.exports.testdaterangefmt = {
     setUp: function(callback) {
         ilib.clearCache();
 
-        length = ["full", "long", "medium", "short"];
-        abbrLength = ["f", "l", "m", "s"];
-
-        fmt = [];
-        templatec00 = [];
-        templatec01 = [];
-        templatec02 = [];
-        templatec03 = [];
-        templatec10 = [];
-        templatec11 = [];
-        templatec12 = [];
-        templatec20 = [];
-        templatec30 = [];
-        
         callback();
     },
     testDateRngFmt_ar_EG: function(test) {
         test.expect(36);
 
-        for (i=0; i < 4; i++) {
-            fmt[i] = new DateRngFmt({locale:"ar-EG", length: length[i]})
-            templatec00[i] = fmt[i].dateFmt.formats.range["c00"][abbrLength[i]];
-            templatec01[i] = fmt[i].dateFmt.formats.range["c01"][abbrLength[i]];
-            templatec02[i] = fmt[i].dateFmt.formats.range["c02"][abbrLength[i]];
-            templatec03[i] = fmt[i].dateFmt.formats.range["c03"][abbrLength[i]];
-            templatec10[i] = fmt[i].dateFmt.formats.range["c10"][abbrLength[i]];
-            templatec11[i] = fmt[i].dateFmt.formats.range["c11"][abbrLength[i]];
-            templatec12[i] = fmt[i].dateFmt.formats.range["c12"][abbrLength[i]];
-            templatec20[i] = fmt[i].dateFmt.formats.range["c20"][abbrLength[i]];
-            templatec30[i] = fmt[i].dateFmt.formats.range["c30"][abbrLength[i]];
+        var length = [], abbrLength = [], fmt;
+
+        length[0] = "full";
+        length[1] = "long";
+        length[2] = "medium";
+        length[3] = "short";
+
+        abbrLength[0] = "f";
+        abbrLength[1] = "l";
```

Comment:
Make abbrLength a global variable as well

Reply:
There's some restriction on QT/QML for running javascript. (https://doc.qt.io/archives/qt-5.6/qtqml-javascript-hostenvironment.html) I've tested that the global variable doesn't work. Instead, I've added  `function` to set variables, then make every test function call it.